nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/DocumentTypeDescriptor.java

```diff
@@ -130,6 +130,13 @@ public DocumentTypeDescriptor merge(DocumentTypeDescriptor other) {
             }
         }
 
+        // update supertype
+        if ((superTypeName == null || superTypeName.isEmpty()) &&
+            (other.superTypeName != null && !other.superTypeName.isEmpty())) {
```

Comment:
Same comment as in the master PR : use `StringUtils.isEmpty` and `StringUtils.isNotEmpty`.